src/index.spec.ts

```diff
@@ -538,7 +538,7 @@ describe('initialization options', () => {
     expect(client.getStringAssignment(flagKey, 'subject', {}, 'default-value')).toBe('control');
   });
 
-  it('only fetches/does initialization workload once per API key if init is called multiple times concurrently', async () => {
+  it.skip('only fetches/does initialization workload once per API key if init is called multiple times concurrently', async () => {
```

Comment:
So, this test fails, not because of the change made in `index.ts` to only buffer one initialization attempt. It fails because the SDK does not actually support instantiating and using multiple clients with different SDKs. The initialization method does all the work on the EppoJsClient.instance singleton. Thus, when init is called with a second SDK/API Key, it overwrites the config store and other properties of the instance loading the first key. By the time all the inits in this test are queued, the reference to `configurationStore` [here](https://github.com/Eppo-exp/js-client-sdk/pull/158/files#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R502) gets out of sync with `instance.configurationStore` for all but the last SDK key to resolve

Reply:
Ah I see the issue is with multiple _different_ API keys. I agree this is currently unsupported functionality. Consider either removing this or adding a comment on how it's commented out because we don't support this... YET